File: src/main/java/kr/co/bookand/backend/article/domain/ArticleTag.java
package kr.co.bookand.backend.article.domain;

import lombok.*;

import javax.persistence.*;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
public class ArticleTag {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "article_tag_id")
    private Long id;

    private String tag;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "article_id")
    private Article article;
}

File: src/main/kotlin/kr/co/bookand/backend/article/domain/ArticleTag.kt
package kr.co.bookand.backend.article.domain

import javax.persistence.*

@Entity
class ArticleTag(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "article_tag_id")
    var id: Long = 0,
    var tag: String,
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "article_id")
    var article: Article? = null
) {
    fun updateArticle(article: Article) {
        this.article = article
    }
}